### PR TITLE
PostBuild tweaked to allow versioning of the Revit assemblies

### LINKED ISSUE #191: Enable versioning of non-standard BHoM assemblies
#### Description:

This request originates from the fact that we have at least a few assemblies that do not follow the standard BHoM conventions to 100%, for various reasons. One of them is for example Revit_Toolkit, which requires references to RevitAPI.dll that cannot be effectively accessed outside of the Revit context, therefore requiring the API dependent part of the code base to have its own project, namespace etc. It would be great to make versioning work for such assemblies too.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #191

### Test files

1. Switch Revit_Toolkit to branch with the same name as this
2. Build Revit_Toolkit in a few solution configs
3. Build Versioning_Toolkit
4. Check if C:\ProgramData\BHoM\Upgrades\BHoMUpgrader52\Upgrades.json contains `Method.ToNew.TestMethod`

Actual versioning will of course only work in the Revit context - but this PR is meant to enable capturing the attributes from Revit assemblies and putting their owners in Upgrades.json, no more than that.

### Additional comments

**Do not merge Revit_Toolkit branch** - it should be deleted once this PR gets tested & approved!

### CODE REVIEW

File: PostBuild/CopyUpgrades.cs [hunk 1]
```diff
@@ -231,7 +231,7 @@ private static void CollectMethodVersioning(BsonDocument content)
             {
                 foreach (PreviousVersionAttribute attribute in method.GetCustomAttributes<PreviousVersionAttribute>())
                 {
-                    if (attribute != null && attribute.FromVersion == version)
+                    if (attribute != null && attribute.FromVersion == version && toNewContent.All(x => x.Name != attribute.PreviousVersionAsText))
```
Comment (IsakNaslundBh) [inline PostBuild/CopyUpgrades.cs:234]: This I am not sure of. I take that this addition is basically saying "do not add if it has already exists"? 
Firstly, this feels like out of scope of what this PR is meant to do unless I am missing something?
More important though is IMO, this should _probably_ be the other way I round I would have thought, as in if it already exists, it should be replaced by the new versioning coming in.
An example of this could for example be a method that is changed more than once in a milestone. Then ofc this post build needs to collect the latest and not keep a previous already existing versioning key there.

I _think_ this would not matter for installers (@FraserGreenroyd to correct me if I am correct or not here) as they start from a clean slate anyway and rebuild everything while being produced, but could lead to annoyance and confusion for us devs.

TL;DR: Would replace old content rather than keep the old and ignore new if there is this kind of clash.

@adecler and @FraserGreenroyd good to get you to chip in here as well.

Comment (pawelbaran) [reply PostBuild/CopyUpgrades.cs:234]: I think you are right, I did not think it through well enough for one main reason: this issue had never occurred before. If it did, the postbuild would simply have crashed on the attempt to add the same element twice. This is what started happening when I was working on this PR, caused by Revit_Core_Engine being built per each version of Revit, resulting in multiple same named classes/methods that need to be versioned.

So I think the right approach would be to overwrite the previous version, as you suggest.

Comment (pawelbaran) [reply PostBuild/CopyUpgrades.cs]: Sure thing @FraserGreenroyd, happy to put as many comments as required

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check copyright-compliance
@BHoMBot check ready-to-merge

**Review by FraserGreenroyd — APPROVED** [review]
Have run through the tests and this works as described happily with the other updates sorted so am happy to see this merged.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 10522414242
